python/sglang/srt/managers/schedule_batch.py

```diff
@@ -543,6 +543,8 @@ def __init__(
 
         # For retraction
         self.is_retracted = False
+        # Indicates if the req has ever been retracted.
+        self.has_retracted = False
```

Comment:
do we really need to introduce this? can't we just use `is_retracted`?

Reply:
@hnyls2002 can you take a look here to see what might be the best way for this? `has_retracted` and `is_retracted` mixing is definitely confusing